sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from "@nestjs/common";
+import {
+  CASAuthDetails,
+  CASSupplierResponse,
+} from "../../processors/schedulers/cas-integration/models/cas-supplier-response.dto";
+import { AxiosRequestConfig } from "axios";
+import { HttpService } from "@nestjs/axios";
+import { InjectRepository } from "@nestjs/typeorm";
+import { CASSupplier } from "@sims/sims-db";
+import { CASIntegrationConfig, ConfigService } from "@sims/utilities/config";
+import { Repository } from "typeorm";
+
+@Injectable()
+export class CASService {
+  private readonly casIntegrationConfig: CASIntegrationConfig;
+  constructor(
+    config: ConfigService,
+    private readonly httpService: HttpService,
+    @InjectRepository(CASSupplier)
+    private readonly casSupplierRepo: Repository<CASSupplier>,
+  ) {
+    this.casIntegrationConfig = config.casIntegration;
+  }
+  /**
+   * Request to login on CAS API and return CAS auth details with the token used for authentication in all other requests.
+   * @returns CAS auth details.
+   */
+  public async casLogon(): Promise<CASAuthDetails> {
```

Comment:
Not a blocker, suggestion, `cas` as prefix would not be needed is a service already named as `CAS`.